frontend/benefit/applicant/src/components/applications/forms/application/step1/extended.ts

```diff
@@ -0,0 +1,130 @@
+import { useTranslation } from 'benefit/applicant/i18n';
+import { FormikProps, useFormik } from 'formik';
+import { TFunction } from 'next-i18next';
+import React, { FormEvent, useState } from 'react';
+import { Field } from 'shared/components/forms/fields/types';
+
+import { APPLICATION_FIELDS } from '../../../constants';
+// import * as Yup from 'yup';
+
+type ExtendedComponentProps = {
+  t: TFunction;
+  fieldNames: string[];
+  fields: FieldsDef;
+  translationsBase: string;
+  getErrorMessage: (fieldName: string) => string | undefined;
+  handleSubmit: (e: FormEvent<HTMLFormElement>) => void;
+  formik: FormikProps<FormFields>;
+};
+
+type FieldsDef = {
+  [key: string]: Field;
+};
+
+type FormFields = {
+  [APPLICATION_FIELDS.HAS_COMPANY_OTHER_ADDRESS]: boolean;
+  [APPLICATION_FIELDS.COMPANY_OTHER_ADDRESS_STREET]: string;
+  [APPLICATION_FIELDS.COMPANY_OTHER_ADDRESS_ZIP]: string;
+  [APPLICATION_FIELDS.COMPANY_OTHER_ADDRESS_DISTRICT]: string;
+  [APPLICATION_FIELDS.COMPANY_IBAN]: string;
+  [APPLICATION_FIELDS.CONTACT_PERSON_FIRST_NAME]: string;
+  [APPLICATION_FIELDS.CONTACT_PERSON_LAST_NAME]: string;
+  [APPLICATION_FIELDS.CONTACT_PERSON_PHONE]: string;
+  [APPLICATION_FIELDS.CONTACT_PERSON_EMAIL]: string;
+  [APPLICATION_FIELDS.DE_MINIMIS_AIDS_GRANTED]: string;
+  [APPLICATION_FIELDS.COLLECTIVE_BARGAINING_ONGOING]: string;
+  [APPLICATION_FIELDS.COLLECTIVE_BARGAINING_INFO]: string;
+};
+
+const useComponent = (): ExtendedComponentProps => {
```

Comment:
There is also some logic that could be extracted to common hook like useStep or something that takes the applicaton fields as parameter and returns common  functionality. But it can be done when starting doing second step.